components/support/jwcrypto/src/encdec.rs

```diff
@@ -0,0 +1,142 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::{EncryptorDecryptorError, JwCryptoError, Jwk};
+use serde::{de::DeserializeOwned, Serialize};
+use std::marker::PhantomData;
+
+/// High-level struct for handling Encryption/Decryption
+///
+/// This struct wraps the jwcrypto functionality in a convenient package.  Also, it uses
+/// `EncryptorDecryptorError`, which includes a description that can help track down where
+/// encryption errors are happening.
+///
+/// EncryptorDecryptor takes a generic Error parameter that it uses for its results.  This can be
+/// anything that implement `From<EncryptorDecryptorError>`, typically it's the internal error type
+/// for a crate.
+pub struct EncryptorDecryptor<E = EncryptorDecryptorError> {
+    jwk: Jwk,
+    phantom: PhantomData<E>,
```

Comment:
nit: do we think https://doc.rust-lang.org/std/marker/struct.PhantomData.html#ownership-and-the-drop-check applies here?

Reply:
TIL